synapseclient/__main__.py

```diff
@@ -550,20 +561,30 @@ def build_parser():
             help='Synapse ID of project or folder where file/folder will be moved ')
     parser_mv.set_defaults(func=move)
 
-    # parser_cp = subparsers.add_parser('cp',
-    #         help='Copies a file in Synapse')
-    # parser_cp.add_argument('--id', metavar='syn123', type=str, required=True,
-    #         help='Id of entity in Synapse to be copied.')
-    # parser_cp.add_argument('--parentid', '--parentId', '-parentid', '-parentId', metavar='syn123', type=str, required=True, dest='parentid',
-    #         help='Synapse ID of project or folder where file will be moved ')
-    # parser_cp.add_argument('-v', '--version', metavar='VERSION', type=int, default=None,
-    #         help='Synapse version number of entity to retrieve. Defaults to most recent version.')
-    # parser_cp.add_argument('--setProvenance', metavar='MODE', type=str, default='traceback',
-    #         help=('Has three values to set the provenance of the copied entity-'
-    #                     'traceback: Sets to the source entity'
-    #                     'existing: Sets to source entity\'s original provenance (if it exists)'
-    #                     'None/none: No provenance is set'))
-    # parser_cp.set_defaults(func=copy)
+    parser_cp = subparsers.add_parser('cp',
+            help='Copies a file in Synapse')
+    parser_cp.add_argument('id', metavar='syn123', type=str,
+            help='Id of entity in Synapse to be copied.')
+    parser_cp.add_argument('--destinationId', metavar='syn123', type=str,
+            help='Synapse ID of project or folder where file will be copied to.  If no destinationId specified, a new project is created')
+    parser_cp.add_argument('--version','-v', metavar='1', type=int, default=None,
+            help='Synapse version number of entity to retrieve. Defaults to most recent version.')
+    parser_cp.add_argument('--mapping', metavar='{test:testing}', type=dict, default=dict(),
+            help="Takes a mapping {'oldSynId': 'newSynId'} to help with replacing syn ids in the new wiki pages that aren't part of the entities being copied. Defaults to dict()")
+    parser_cp.add_argument('--setProvenance', metavar='traceback', type=str, default='traceback',
+            help=('Has three values to set the provenance of the copied entity-'
+                        'traceback: Sets to the source entity'
+                        'existing: Sets to source entity\'s original provenance (if it exists)'
+                        'None/none: No provenance is set'))
+    parser_cp.add_argument('--skipCopyTable', action='store_false',
+            help='Skips Table Entities when copying projects')
+    parser_cp.add_argument('--replace', action='store_true',
+            help='Replace File Entities even if there is a file entities that are named the same in the destination')
+    parser_cp.add_argument('--notRecursive', action='store_false',
+            help='Do not copy everything recusively')
```

Comment:
This does not represents what this variable does.  It might be better to have options excludeTypes=('folder')

excludeTypes=(, 'table')

enum of file, table, wiki